File: server/task-service/model/task.go
package model

import (
	"encoding/json"
	"go.mongodb.org/mongo-driver/bson/primitive"
	"io"
	"time"
)

type TaskStatus string

const (
	Pending    TaskStatus = "Pending"
	InProgress TaskStatus = "InProgress"
	Completed  TaskStatus = "Completed"
)

type Task struct {
	ID           primitive.ObjectID `bson:"_id,omitempty" json:"id"`
	ProjectID    string             `bson:"project_id" json:"projectId"`
	Name         string             `bson:"name" json:"name"`
	Description  string             `bson:"description" json:"description"`
	Status       TaskStatus         `bson:"status" json:"status"`
	CreatedAt    time.Time          `bson:"created_at" json:"createdAt"`
	UpdatedAt    time.Time          `bson:"updated_at" json:"updatedAt"`
	UserIDs      []string           `bson:"user_ids" json:"user_ids"`
	Dependencies []string           `bson:"dependencies" json:"dependencies"`
	Blocked      bool               `bson:"blocked" json:"blocked"`
}

func (t *Task) ToJSON(w io.Writer) error {
	e := json.NewEncoder(w)
	return e.Encode(t)
}

func (t *Task) FromJSON(r io.Reader) error {
	d := json.NewDecoder(r)
	return d.Decode(t)
}

File: client/ui-trello/src/app/models/task.ts
export type TaskStatus = 'Pending' | 'In Progress' | 'Completed';

export class Task {
  id: string;
  projectId: string;
  name: string;
  description: string;
  status: TaskStatus = 'Pending';
  createdAt: Date;
  updatedAt: Date;
  user_ids: string[];
  dependencies: string[];
  blocked: boolean;

  constructor(
    id: string,
    projectId: string,
    name: string,
    description: string,
    status: TaskStatus = 'Pending',
    createdAt: Date,
    updatedAt: Date,
    user_ids: string[] = [],
    dependencies: string[] = [],
    blocked: boolean = false
  ) {
    this.id = id;
    this.projectId = projectId;
    this.name = name;
    this.description = description;
    this.status = status;
    this.createdAt = createdAt;
    this.updatedAt = updatedAt;
    this.user_ids = user_ids;
    this.dependencies = dependencies;
    this.blocked = blocked;
  }
}
